operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java

```diff
@@ -67,6 +72,38 @@ public void testWorkingRealmImport() {
         });
     }
 
+    @Test
+    @EnabledIfSystemProperty(named = OPERATOR_CUSTOM_IMAGE, matches = ".+")
+    public void testWorkingRealmImportWithCustomImage() {
+        // Arrange
+        // Recreating the database and the realm import CR to keep this test isolated
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).delete();
+        deleteDB();
+        deployDB();
+        var keycloak = getDefaultKeycloakDeployment();
+        keycloak.getSpec().setImage(customImage);
+        deployKeycloak(k8sclient, keycloak, false);
+
+        // Act
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).createOrReplace();
+
+        // Assert
```

Comment:
Should we maybe add a similar assert to this to check that build is not present?
https://github.com/keycloak/keycloak/blob/5289b8704de36e6bfc9edcecffa7b639052465cb/operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java#L344-L345
WDYT?

Reply:
also wanted to ask if this tests really checks if the build does not take place when a custom image is given. as i can see it it is just checking if the import takes place, but not if the build actually happens or not, right? so this test would also be green when a build takes place as of now without the additional assert.

Reply:
something on those lines 👍  ok